test/run.sh

```diff
@@ -1,77 +1,49 @@
 #!/bin/bash
 
-dnsmasq_pre() {
-  echo -n $(date +"%b %e %H:%M:%S")
-  echo -n "dnsmasq[123]:"
-}
+# Only run tests on x86_64 target
+if [[ "${1}" != "pihole-FTL-linux-x86_64" ]]; then
+  echo "Skipping tests (${1})!"
+  exit 0
+fi
 
-# Prepare FTL's files
-ts="$(dnsmasq_pre)"
-cat <<EOT >> pihole.log
-${ts} 1 1270.0.01/1234 query[AAAA] raspberrypi from 127.0.0.1
-${ts} 1 1270.0.01/1234 /etc/pihole/local.list raspberrypi is fda2:2001:5647:0:ba27:ebff:fe37:4205
-${ts} 2 1270.0.01/1234 query[A] ChEcKiP.DyNdNs.OrG from 127.0.0.1
-${ts} 2 1270.0.01/1234 forwarded ChEcKiP.DyNdNs.OrG to 2001:1608:10:25::9249:d69b
-${ts} 2 1270.0.01/1234 forwarded ChEcKiP.DyNdNs.OrG to 2001:1608:10:25::1c04:b12f
-${ts} 2 1270.0.01/1234 forwarded ChEcKiP.DyNdNs.OrG to 2620:0:ccd::2
-${ts} 2 1270.0.01/1234 forwarded ChEcKiP.DyNdNs.OrG to 2620:0:ccc::2
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is <CNAME>
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 216.146.38.70
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 216.146.43.71
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 91.198.22.70
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 216.146.43.70
-${ts} 3 1270.0.01/1234 query[A] pi.hole from 10.8.0.2
-${ts} 3 1270.0.01/1234 /etc/pihole/local.list pi.hole is 192.168.2.10
-${ts} 4 1270.0.01/1234 query[A] example.com from 10.8.0.2
-${ts} 4 1270.0.01/1234 /etc/pihole/local.list example.com is 192.168.2.10
-${ts} 5 1270.0.01/1234 query[A] play.google.com from 192.168.2.208
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2001:1608:10:25::9249:d69b
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2001:1608:10:25::1c04:b12f
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2620:0:ccd::2
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2620:0:ccc::2
-${ts} 5 1270.0.01/1234 reply play.google.com is <CNAME>
-${ts} 5 1270.0.01/1234 reply play.l.google.com is 216.58.208.110
-${ts} 5 1270.0.01/1234 reply play.l.google.com is 216.58.208.110
-${ts} 5 1270.0.01/1234 reply play.l.google.com is 216.58.208.110
-${ts} 5 1270.0.01/1234 reply play.google.com is <CNAME>
-${ts} 6 1270.0.01/1234 query[AAAA] play.google.com from 192.168.2.208
-${ts} 6 1270.0.01/1234 forwarded play.google.com to 2620:0:ccd::2
-${ts} 6 1270.0.01/1234 reply play.l.google.com is 2a00:1450:4017:802::200e
-${ts} 7 1270.0.01/1234 query[A] blacklisted.com from 192.168.2.208
-${ts} 7 1270.0.01/1234 /etc/pihole/black.list blacklisted.com is 1.2.3.4
-${ts} 8 1270.0.01/1234 query[A] addomain.com from 192.168.2.208
-${ts} 8 1270.0.01/1234 /etc/pihole/gravity.list addomain.com is 1.2.3.4
-EOT
-touch "pihole-FTL.log"
+# Install necessary additional components for testing
+apt-get -qq install dnsutils -y > /dev/null
 
-cat <<EOT >> pihole-FTL.conf
-DBFILE=pihole-FTL.db
-LOGFILE=pihole-FTL.log
-SOCKETFILE=pihole-FTL.sock
-EOT
+# Create necessary directories
+mkdir -p /etc/pihole /var/run/pihole /var/log
+
+# Download prepared gravity database
+wget https://ftl.pi-hole.net/gravity.db
```

Comment:
Please change this to a SQL file in this repo which is used to generate the test database. There are a lot of downsides to hosting a test database like this:
- It requires us to remember to keep a database in ftl.pi-hole.net, which is not meant for hosting test data
- It is not version controlled
- The database as it is now is not reproducible.
- If the tests or database code are modified in a feature branch in a way that is incompatible with the hosted test database, the branch will fail the build because of the incompatibility.
- The tests are not reproducible in an offline system.